lib/components/Upload.js

```diff
@@ -181,6 +181,7 @@ export default class Upload extends Component {
           disabled={disabled}
           ref="file"
           type="file"
+          accept={upload.source.extension}
```

Comment:
Probably should use `_.get` for some extra security on the availability of the `extension` field. We wouldn't want to crash the whole app just because we can't find that property. Although then we need a fallback for the file extension...is there a wildcard you can (pointlessly) give to `accept`?